--- server.rb.orig	2006-04-11 19:50:28.000000000 +0200
+++ server.rb	2006-04-11 19:50:11.000000000 +0200
@@ -12,6 +12,8 @@
     ARGV.shift
   when "webrick"
     ARGV.shift
+  when "webrick-ssl"
+    ARGV.shift
   else
     if RUBY_PLATFORM !~ /mswin/ && !silence_stderr { `lighttpd -version` }.blank? && defined?(FCGI)
       "lighttpd"
@@ -22,6 +24,8 @@
 
 if server == "webrick"
   puts "=> Booting WEBrick..."
+elsif server == "webrick-ssl"
+  puts "=> Booting WEBrick-ssl..."
 else
   puts "=> Booting lighttpd (use 'script/server webrick' to force WEBrick)"
 end
